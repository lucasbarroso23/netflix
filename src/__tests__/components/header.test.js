import React from 'react';
import { render, fireEvent } from '@testing-library/react';
import { Header } from '../../components';

jest.mock('react-router-dom');

describe('<Header />', () => {
  it('render <Header /> with populated values', () => {
    const { container, getByText, getByAltText, getByTestId } = render(
      <Header>
        <Header.Frame>
          <Header.Logo src="/logo.svg" alt="Netflix" />
          <Header.TextLink active="true">Hello I am a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    )

    expect(getByText('Hello I am a link!')).toBeTruthy();
    expect(getByTestId('header-bg')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  })
})
