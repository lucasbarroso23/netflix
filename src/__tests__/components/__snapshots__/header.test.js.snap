// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header /> render the basic <Header /> with a background 1`] = `
<div
  class="header__Background-djltcn-0 lglpeb"
  data-testid="header-bg"
>
  <div
    class="header__Container-djltcn-1 gnNVjt"
  >
    <p
      class="header__Link-djltcn-6 hdYwTX"
    >
      Hello I am a link!
    </p>
  </div>
</div>
`;

exports[`<Header /> render the basic <Header /> without a background 1`] = `
<div
  class="header__Container-djltcn-1 gnNVjt"
>
  <p
    class="header__Link-djltcn-6 ktcrzT"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`<Header /> renders the full <Header> with a background 1`] = `
<div
  class="header__Background-djltcn-0 dUzWEZ"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-djltcn-1 gnNVjt"
  >
    <div
      class="header__Group-djltcn-8 bFaOer"
    >
      <p
        class="header__Link-djltcn-6 ktcrzT"
      >
        Series
      </p>
      <p
        class="header__Link-djltcn-6 ktcrzT"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-djltcn-8 bFaOer"
    >
      <div
        class="header__Search-djltcn-11 fbyXKp"
      >
        <button
          class="header__SearchIcon-djltcn-12 jDgbuc"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-djltcn-14 gpMqHq"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-djltcn-15 jQomNN"
      >
        <button
          class="header__Picture-djltcn-9 gIHqVJ"
          src="/images/users//images/karl.png.png"
        />
        <div
          class="header__Dropdown-djltcn-10 blIbwk"
        >
          <div
            class="header__Group-djltcn-8 bFaOer"
          >
            <button
              class="header__Picture-djltcn-9 gIHqVJ"
              src="/images/users//images/karl.png.png"
            />
            <p
              class="header__Link-djltcn-6 ktcrzT"
            >
              Karl Hadwen
            </p>
          </div>
          <div
            class="header__Group-djltcn-8 bFaOer"
          >
            <p
              class="header__Link-djltcn-6 ktcrzT"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-djltcn-1 header__Feature-djltcn-4 gnNVjt hiaCZx"
  >
    <h2
      class="header__FeatureCallOut-djltcn-7 geVrgU"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-djltcn-5 zHXyP"
    >
      Forever alone in a crowd...
    </p>
    <button
      class="header__PlayButton-djltcn-13 hRbbto"
    >
      Play
    </button>
  </div>
</div>
`;
